src/features/courses/course-controller.ts

```diff
@@ -2857,33 +2871,130 @@ class CourseController {
         const {
             questionId,
             userId,
-            includeWorkbooks
+            includeWorkbooks,
+            userRole,
         } = options;
-        const include: sequelize.IncludeOptions[] = [{
-            model: StudentGradeInstance,
-            as: 'gradeInstances',
-            required: false,
+
+        const where: sequelize.WhereOptions = {
+            courseWWTopicQuestionId: questionId,
+            userId,
+            active: true,
+        };
+
+        const topicAssessmentInfo = await TopicAssessmentInfo.findOne({
+            attributes: [ 'showItemizedResults', 'showTotalGradeImmediately' ],
             where: {
                 active: true,
+            },
+            include: [
+                {
+                    model: CourseTopicContent,
+                    as: 'courseTopicContent',
+                    attributes: [ 'id' ],
+                    required: true,
+                    where: {
+                        active: true,
+                    },
+                    include: [
+                        {
+                            model:  CourseWWTopicQuestion,
+                            as: 'questions',
+                            attributes: [],
+                            required: true,
+                            where: {
+                                active: true,
+                                id: questionId,
+                            }
+                        }
+                    ]
+                }
+            ]
+        });
+
+        let attributes;
+        let gradeInstanceAttributes;
+        let workbookAttributes;
+        if (userRole === Role.STUDENT && (topicAssessmentInfo?.showTotalGradeImmediately === false || topicAssessmentInfo?.showItemizedResults === false)) {
```

Comment:
Shouldn't this be

userRole is student AND
showTotalGradeImmediately AND
showItemizedResults

The or with the falses seems off

Reply:
or maybe it should just be that they are true? unless there is a double negative somewhere?

Reply:
Oh unless you're using this to filter the attributes and relying on them being undefined if not entered here to show everything?

Without types on the variables it's pretty confusing